packages/loader/container-loader/src/container.ts

```diff
@@ -833,10 +833,31 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             this._resolvedUrl = resolvedUrl;
             await this.connectStorageService();
 
-            // upload blobs here (NYI)
-
-            // post summary here
             if (hasAttachmentBlobs) {
+                // upload blobs to storage
+                assert(!!this.loader.services.detachedBlobStorage, "assertion for type narrowing");
+                const redirectTable: Map<string, string | undefined>
+                    = new Map(this.loader.services.detachedBlobStorage.all().map((id) => [id, undefined]));
+                do {
+                    for (const id of this.loader.services.detachedBlobStorage.all()) {
```

Comment:
Can we do all() call less frequently? Like at the end of the loop and only if there are no more blobs to work with?